lib/dependencies/HarmonyTopLevelThisParserPlugin.js

```diff
@@ -0,0 +1,25 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Florent Cailhol @ooflorent
+*/
+"use strict";
+
+const ConstDependency = require("./ConstDependency");
+
+class HarmonyTopLevelThisParserPlugin {
+	apply(parser) {
+		parser.hooks.expression.for("this").tap("HarmonyTopLevelThisParserPlugin", node => {
+			if(!parser.scope.topLevelScope)
+				return;
+			const module = parser.state.module;
+			const isHarmony = !!(module.buildMeta && module.buildMeta.exportsType);
+			if(isHarmony) {
+				const dep = new ConstDependency("(void 0)", node.range, false);
```

Comment:
Replace `(void 0)` with `undefined`.

Inserting brackets is not safe because of ASI.

``` js
something
this
```

After replacing and ASI this will call the `something` variable. `something(void 0)`

Reply:
Fixed.